tests/integration_tests/functional/test_balloon.py

```diff
@@ -0,0 +1,713 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""Tests for guest-side operations on /balloon resources."""
+
+import logging
+import os
+import platform
+import subprocess
+import time
+
+import pytest
+
+from conftest import _test_images_s3_bucket
+from framework.artifacts import ArtifactCollection, ArtifactSet
+from framework.builder import MicrovmBuilder, SnapshotBuilder, SnapshotType
+from framework.matrix import TestMatrix, TestContext
+from framework.utils import get_free_mem_ssh
+
+import host_tools.network as net_tools  # pylint: disable=import-error
+
+
+MB_TO_PAGES = 256
+
+
+def get_rss_mem_by_pid(pid):
+    """Get the RSS memory that a guest uses, given the pid of the guest."""
+    output = subprocess.check_output("pmap -X {}".format(pid), shell=True)
+    return int(output.decode('utf-8').split('\n')[-2].split()[1], 10)
+
+
+def make_guest_dirty_memory(ssh_connection, should_oom=False, amount=8192):
+    """Tell the guest, over ssh, to dirty `amount` pages of memory."""
+    amount_in_mbytes = amount / MB_TO_PAGES
+
+    exit_code, _, _ = ssh_connection.execute_command(
+        "/sbin/fillmem {}".format(amount_in_mbytes)
+    )
+    time.sleep(0.1)
+
+    cmd = "cat /tmp/fillmem_output.txt"
```

Comment:
Looks like this could be more deterministic if you only created and filled the file (in the C code) *after* you run the memory load, and you wait here for the file to show up on disk and be non-zero-sized.

Reply:
I will just remove it altogether as the program should have written every byte and closed the fd by the time control is returned in the test from the ssh command.

Reply:
Removed it.